File: test-junit/src/test/java/io/micronaut/test/junit/RequiresTest.java

package io.micronaut.test.junit;

import io.micronaut.context.annotation.Requires;
import io.micronaut.test.extensions.junit.annotation.MicronautTest;
import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

@MicronautTest
@Requires(property = "does.not.exist")
class RequiresTest {

    @Test
    void testNotExecuted() {
        Assertions.fail("Should never be executed");
    }
}

File: test-suite-python/src/test/python/micronaut/test/python/RequiresTest.py
from micronaut.context.annotation import Requires
from micronaut.test.extensions.junit5.annotation import MicronautTest
from org.junit.jupiter.api import Test


@MicronautTest
@Requires(property="does.not.exist")
class RequiresTest:

    @Test
    def test_not_executed(self):
        assert False, "Should never be executed"
